writeengine/wrapper/writeengine.cpp

```diff
@@ -6173,42 +6160,38 @@ void WriteEngineWrapper::RemoveTxnFromDictMap(const TxnID txnid)
     }
 }
 
+int WriteEngineWrapper::markTxnExtentsAsInvalid(const TxnID txnid, bool erase)
+{
+    int rc = 0;
+    std::tr1::unordered_map<TxnID, SP_TxnLBIDRec_t>::iterator mapIter = m_txnLBIDMap.find(txnid);
+    if (mapIter != m_txnLBIDMap.end())
+    {
+        SP_TxnLBIDRec_t spTxnLBIDRec = (*mapIter).second;
+        if (spTxnLBIDRec->m_LBIDs.size())
+        {
+            rc = BRMWrapper::getInstance()->markExtentsInvalid(spTxnLBIDRec->m_LBIDs, spTxnLBIDRec->m_ColTypes);
+        }
+            
+        if (erase)
+        {
+            m_txnLBIDMap.erase(txnid);   // spTxnLBIDRec is auto-destroyed
+        }
+    }
+    return rc;
+}
+
 /***********************************************************
  * DESCRIPTION:
  *    Remove a transaction LBID list from the LBID map
  *    Called when a transaction ends, either commit or rollback
  * PARAMETERS:
  *    txnid - the transaction to remove.
  * RETURN:
- *    0 => success or not found, -1 => error
+ *    error code
  ***********************************************************/
 int WriteEngineWrapper::RemoveTxnFromLBIDMap(const TxnID txnid)
 {
-    int rtn = 0;
-    std::tr1::unordered_map<TxnID, SP_TxnLBIDRec_t>::iterator mapIter;
-
-    // Find the set of extent starting LBIDs for this transaction. If not found, then create it.
-    try
-    {
-        mapIter = m_txnLBIDMap.find(txnid);
-
-        if (mapIter != m_txnLBIDMap.end())
-        {
-            SP_TxnLBIDRec_t spTxnLBIDRec = (*mapIter).second;
-            // Debug
-//            cout << "Remove transaction entry " << txnid << " transaction count " << m_txnLBIDMap.size() << endl;
-//            cout << "    count = " << spTxnLBIDRec->m_LBIDMap.size() <<
-//                    ", lastSeqnum = " <<  spTxnLBIDRec->m_lastSeqnum <<
-//                    ", squashed lbids = " << spTxnLBIDRec->m_squashedLbids << endl;
-            m_txnLBIDMap.erase(txnid);   // spTxnLBIDRec is auto-destroyed
-        }
-    }
-    catch (...)
-    {
-        rtn = -1;
-    }
-
-    return rtn;
+    return markTxnExtentsAsInvalid(txnid, true);
```

Comment:
`RemoveTxnFromLBIDMap` is called for transaction rollbacks as well. So I don't think marking the extents as invalid for a rolled-back transaction is a desired behaviour here. This function should only be erasing the entry from `m_txnLBIDMap` for a given `txnid`.

Reply:
This is an expected behavior. We will be marking extents invalid on rollback.